### PR TITLE
Document overflow methods

### PR DESCRIPTION
**Pull request by kylekatarnls**
Documentation for:
- static useMonthsOverflow
- static resetMonthsOverflow
- static shouldOverflowMonths
- addMonthsWithOverflow
- addMonthWithOverflow
- addMonthsNoOverflow
- addMonthNoOverflow
- subMonthsWithOverflow
- subMonthWithOverflow
- subMonthsNoOverflow
- subMonthNoOverflow

### CODE REVIEW

**Comment by kylekatarnls** [comment]
Shit, my locale is taken into account for the auto-generation. I will add somthing for that in the generate.php script.

File: docs/index.html [hunk 1]
```diff
@@ -179,8 +179,8 @@ <h1 id="api-instantiation">Instantiation</h1>
 
 <p>Ever need to loop through some dates to find the earliest or latest date?  Didn't know what to set your initial maximum/minimum values to? There are now two helpers for this to make your decision simple:</p>
 
-<p><pre><code class="php">echo Carbon::maxValue();                               // '2038-01-18 22:14:07'
-echo Carbon::minValue();                               // '1901-12-13 15:45:52'
+<p><pre><code class="php">echo Carbon::maxValue();                               // '9999-12-31 23:59:59'
```
Comment (Glavic) [inline docs/index.html]: That really depends on system OS (32 or 64bit).
https://en.wikipedia.org/wiki/Year_2038_problem
Should we also comment that?

File: docs/index.html [hunk 2]
```diff
@@ -225,10 +225,12 @@ <h1 id="api-testing">Testing Aids</h1>
 echo Carbon::parse();                                  // 2001-05-21 12:00:00
 echo new Carbon('now');                                // 2001-05-21 12:00:00
 echo Carbon::parse('now');                             // 2001-05-21 12:00:00
-var_dump(Carbon::hasTestNow());                        // bool(true)
+var_dump(Carbon::hasTestNow());                        // C:\Users\bastien.miclo\Perso\Carbon\generate.php(44) : eval()'d code:1:
```
Comment (Glavic) [inline docs/index.html]: `C:\Users\bastien.miclo\Perso\Carbon\generate.php(44)` should that be here?

File: docs/index.html (hunk 1 above)
Comment (kylekatarnls) [reply docs/index.html]: I was exactly thinking of that. :)

File: docs/index.html [hunk 3]
```diff
@@ -225,10 +225,12 @@ <h1 id="api-testing">Testing Aids</h1>
...
+bool(true)
 Carbon::setTestNow();                                  // clear the mock
-var_dump(Carbon::hasTestNow());                        // bool(false)
-echo Carbon::now();                                    // 2016-06-24 15:18:34
+var_dump(Carbon::hasTestNow());                        // C:\Users\bastien.miclo\Perso\Carbon\generate.php(44) : eval()'d code:1:
```
Comment (Glavic) [inline docs/index.html]: ... and here?

File: docs/index.html (hunk 2 above)
Comment (kylekatarnls) [reply docs/index.html]: Shit, xdebug must be disabled for the script.

File: docs/index.html [hunk 4]
```diff
@@ -426,7 +458,7 @@ <h1 id="api-commonformats">Common Formats</h1>
 // $dt->toAtomString() is the same as $dt->format(DateTime::ATOM);
 echo $dt->toAtomString();      // 1975-12-25T14:15:16-05:00
 echo $dt->toCookieString();    // Thursday, 25-Dec-1975 14:15:16 EST
-echo $dt->toIso8601String();   // 1975-12-25T14:15:16-0500
+echo $dt->toIso8601String();   // 1975-12-25T14:15:16-05:00
```
Comment (Glavic) [inline docs/index.html]: Before that, we should investigate: https://github.com/briannesbitt/Carbon/pull/977

**Comment by kylekatarnls** [comment]
I cleaned it a bit but I must fix this: https://github.com/briannesbitt/Carbon/pull/1088/files#diff-eae600270579316beb8b8d9b9a5732d4L189

**Comment by kylekatarnls** [comment]
In fact, the example assumes the default locale is en, however it depends on the system. So I will rather change the example.

**Comment by kylekatarnls** [comment]
min and max documented according to 32/64 bits conditional

File: docs/index.src.html [hunk 5]
```diff
@@ -136,14 +136,21 @@ <h1 id="api-instantiation">Instantiation</h1>
 {{minValue::exec(echo Carbon::minValue();/*pad(54)*/)}} // '{{minValue_eval}}'
 </code></pre></p>
 
+<p>Min and max value mainly depends on the system (32 or 64 bits).</p>
```
Comment (briannesbitt) [inline docs/index.src.html]: The min and max values depend on the bit count of the system (32 or 64 bit).

File: docs/index.src.html [hunk 6]
```diff
@@ -136,14 +136,21 @@ <h1 id="api-instantiation">Instantiation</h1>
...
+
+<p>With a 32 bits OS system (you can check it in PHP with <code>PHP_INT_SIZE === 4</code>), the minimum value is the 0-unix-timestamp (1970-01-01 00:00:00) and the maximum is the timestamp given by the constant <code>PHP_INT_MAX</code>.</p>
```
Comment (briannesbitt) [inline docs/index.src.html]: With a 32-bit OS system (you can check it in PHP using <code>PHP_INT_SIZE === 4</code>), the minimum value is 0-unix-timestamp (1970-01-01 00:00:00) and the maximum is the timestamp given by the constant <code>PHP_INT_MAX</code>.

File: docs/index.src.html [hunk 7]
```diff
@@ -136,14 +136,21 @@ <h1 id="api-instantiation">Instantiation</h1>
...
+
+<p>With a 64 bits OS system, the minimum is 01-01-01 00:00:00 and maximum is 9999-12-31 23:59:59.</p>
```
Comment (briannesbitt) [inline docs/index.src.html]: With a 64-bit OS system, the minimum is 01-01-01 00:00:00 and maximum is 9999-12-31 23:59:59.

File: docs/index.src.html [hunk 8]
```diff
@@ -532,6 +539,61 @@ <h1 id="api-addsub">Addition and Subtraction</h1>
 
 <p>P.S. Don't worry if you forget and use <code>addDay(5)</code> or <code>subYear(3)</code>, I have your back ;)</p>
 
+<p>By default, Carbon behave as the PHP DateTime do. That means adding or substracting months can overflow, example:</p>
```
Comment (briannesbitt) [inline docs/index.src.html]: By default, Carbon relies on the underlying parent class PHP DateTime behavior.  As a result adding or subtracting months can overflow, example:

File: docs/index.src.html [hunk 9]
```diff
@@ -532,6 +539,61 @@ <h1 id="api-addsub">Addition and Subtraction</h1>
...
+
+<p><pre><code class="php">{{::lint($dt = Carbon::create(2017, 1, 31, 0);)}}
+
+{{addoverflow1::exec(echo $dt->copy()->addMonth();/*pad(40)*/)}} // {{addoverflow1_eval}}
+{{addoverflow2::exec(echo $dt->copy()->subMonths(2);/*pad(40)*/)}} // {{addoverflow2_eval}}
+</code></pre></p>
+
+<p>You can't prevent the overflow with <code>Carbon::useMonthsOverflow(false)</code></p>
```
Comment (briannesbitt) [inline docs/index.src.html]: You **can** prevent

File: docs/index.src.html [hunk 10]
```diff
@@ -532,6 +539,61 @@ <h1 id="api-addsub">Addition and Subtraction</h1>
...
+$dt = Carbon::create(2017, 1, 31, 0);)}}
+
+{{addoverflow3::exec(echo $dt->copy()->addMonth();/*pad(40)*/)}} // {{addoverflow3_eval}}
+{{addoverflow4::exec(echo $dt->copy()->subMonths(2);/*pad(40)*/)}} // {{addoverflow4_eval}}
+
+// Call the method with true to allow overflow again
+{{reset1::exec(Carbon::resetMonthsOverflow();)}}} // same as Carbon::useMonthsOverflow(true);
+</code></pre></p>
+
+The method <code>Carbon::shouldOverflowMonths()</code> allow you to know if the overflow is currently enabled.
```
Comment (briannesbitt) [inline docs/index.src.html]: allow**s**

**Pull request merged**